Looks like you're still having some style issues. From the CI:

```
$ flake8 homeassistant --exclude bower_components,external
homeassistant/components/switch/mqtt.py:7:80: E501 line too long (87 > 79 characters)
homeassistant/components/switch/mqtt.py:8:79: W291 trailing whitespace
homeassistant/components/switch/mqtt.py:9:80: E501 line too long (88 > 79 characters)
homeassistant/components/switch/mqtt.py:9:89: W291 trailing whitespace
homeassistant/components/switch/mqtt.py:12:80: E501 line too long (80 > 79 characters)
homeassistant/components/switch/mqtt.py:13:80: E501 line too long (80 > 79 characters)
homeassistant/components/switch/mqtt.py:14:80: E501 line too long (94 > 79 characters)
homeassistant/components/switch/mqtt.py:16:65: W291 trailing whitespace
homeassistant/components/switch/mqtt.py:39:52: W291 trailing whitespace
homeassistant/components/switch/mqtt.py:74:1: E302 expected 2 blank lines, found 1
homeassistant/components/switch/mqtt.py:80:1: W293 blank line contains whitespace
homeassistant/components/switch/mqtt.py:90:1: E302 expected 2 blank lines, found 1
homeassistant/components/switch/mqtt.py:92:80: E501 line too long (100 > 79 characters)
homeassistant/components/switch/mqtt.py:93:28: W291 trailing whitespace
homeassistant/components/switch/mqtt.py:101:1: W293 blank line contains whitespace
```
```
$ pylint homeassistant
************* Module homeassistant.components.sensor.mqtt
C: 69, 8: Missing function docstring (missing-docstring)
************* Module homeassistant.components.switch.mqtt
C:  8, 0: Trailing whitespace (trailing-whitespace)
C:  9, 0: Trailing whitespace (trailing-whitespace)
C: 16, 0: Trailing whitespace (trailing-whitespace)
C: 39, 0: Trailing whitespace (trailing-whitespace)
C: 80, 0: Trailing whitespace (trailing-whitespace)
C: 93, 0: Trailing whitespace (trailing-whitespace)
C:101, 0: Trailing whitespace (trailing-whitespace)
C:103, 0: Unnecessary parens after 'if' keyword (superfluous-parens)
C:106, 0: Unnecessary parens after 'elif' keyword (superfluous-parens)
R: 90, 0: Too many instance attributes (8/7) (too-many-instance-attributes)
R: 92, 4: Too many arguments (8/5) (too-many-arguments)
C:102, 8: Missing function docstring (missing-docstring)
W:111,12: String statement has no effect (pointless-string-statement)
W:114,12: String statement has no effect (pointless-string-statement)
W:136,12: String statement has no effect (pointless-string-statement)
W:144,12: String statement has no effect (pointless-string-statement)
```

--- a/homeassistant/components/sensor/mqtt.py
+++ b/homeassistant/components/sensor/mqtt.py
@@ -67,6 +67,7 @@ class MqttSensor(Entity):
         self._unit_of_measurement = unit_of_measurement
 
         def message_received(topic, payload, qos):
+            """ A new MQTT message has been received. """
             self._state = payload
             self.update_ha_state()
 

--- a/homeassistant/components/switch/mqtt.py
+++ b/homeassistant/components/switch/mqtt.py
@@ -4,16 +4,18 @@ homeassistant.components.switch.mqtt
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Allows to configure a MQTT switch.
 
-In an ideal scenario, the MQTT device will have a state topic to publish state changes.
-If these messages are published with RETAIN flag, the MQTT switch will receive 
-an instant state update after subscription and will start with correct state. Otherwise, 
-the initial state of the switch will be false/off.
+In an ideal scenario, the MQTT device will have a state topic to publish
+state changes. If these messages are published with RETAIN flag, the MQTT
+switch will receive an instant state update after subscription and will 
+start with correct state. Otherwise, the initial state of the switch will
+be false/off.
 
-When a state topic is not available, the switch will work in an optimistic mode.
-In this mode, the MQTT switch will immediately change state after every command.
-Otherwise, the switch will wait for state confirmation from device (message from state_topic).
+When a state topic is not available, the switch will work in optimistic mode.
+In this mode, the switch will immediately change state after every command.
+Otherwise, the switch will wait for state confirmation from device
+(message from state_topic).
 
-Optimistic mode can be forced, even if state topic is available. 
+Optimistic mode can be forced, even if state topic is available.
 Try to enable it, if experiencing incorrect switch operation.
 
 
@@ -36,7 +38,7 @@ The name of the switch. Default is 'MQTT Switch'.
 
 state_topic
 *Optional
-The MQTT topic subscribed to receive state updates. 
+The MQTT topic subscribed to receive state updates.
 If not specified, optimistic mode will be forced.
 
 command_topic
@@ -70,6 +72,7 @@ DEFAULT_OPTIMISTIC = False
 
 DEPENDENCIES = ['mqtt']
 
+
 # pylint: disable=unused-argument
 def setup_platform(hass, config, add_devices_callback, discovery_info=None):
     """ Add MQTT Switch """
@@ -77,7 +80,7 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
     if config.get('command_topic') is None:
         _LOGGER.error("Missing required variable: command_topic")
         return False
-    
+
     add_devices_callback([MqttSwitch(
         hass,
         config.get('name', DEFAULT_NAME),
@@ -87,10 +90,12 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
         config.get('payload_off', DEFAULT_PAYLOAD_OFF),
         config.get('optimistic', DEFAULT_OPTIMISTIC))])
 
+
 class MqttSwitch(SwitchDevice):
     """ Represents a switch that can be togggled using MQTT """
-    def __init__(self, hass, name, state_topic, command_topic, payload_on, payload_off, optimistic):
-        self._state = False        
+    def __init__(self, hass, name, state_topic, command_topic,
+                 payload_on, payload_off, optimistic):
+        self._state = False
         self._hass = hass
         self._name = name
         self._state_topic = state_topic
@@ -98,20 +103,21 @@ class MqttSwitch(SwitchDevice):
         self._payload_on = payload_on
         self._payload_off = payload_off
         self._optimistic = optimistic
-            
+
         def message_received(topic, payload, qos):
-            if (payload == self._payload_on):
+            """ A new MQTT message has been received. """
+            if payload == self._payload_on:
                 self._state = True
                 self.update_ha_state()
-            elif (payload == self._payload_off):
+            elif payload == self._payload_off:
                 self._state = False
                 self.update_ha_state()
 
         if self._state_topic is None:
-            """ force optimistic mode """
+            # force optimistic mode
             self._optimistic = True
         else:
-            """ subscribe the state_topic """
+            # subscribe the state_topic
             mqtt.subscribe(hass, self._state_topic, message_received)
 
     @property
@@ -133,7 +139,7 @@ class MqttSwitch(SwitchDevice):
         """ Turn the device on. """
         mqtt.publish(self.hass, self._command_topic, self._payload_on)
         if self._optimistic:
-            """ optimistically assume that switch has changed state """
+            # optimistically assume that switch has changed state
             self._state = True
             self.update_ha_state()
 
@@ -141,6 +147,6 @@ class MqttSwitch(SwitchDevice):
         """ Turn the device off. """
         mqtt.publish(self.hass, self._command_topic, self._payload_off)
         if self._optimistic:
-            """ optimistically assume that switch has changed state """
+            # optimistically assume that switch has changed state
             self._state = False
             self.update_ha_state()

--- a/homeassistant/components/switch/mqtt.py
+++ b/homeassistant/components/switch/mqtt.py
@@ -6,7 +6,7 @@ Allows to configure a MQTT switch.
 
 In an ideal scenario, the MQTT device will have a state topic to publish
 state changes. If these messages are published with RETAIN flag, the MQTT
-switch will receive an instant state update after subscription and will 
+switch will receive an instant state update after subscription and will
 start with correct state. Otherwise, the initial state of the switch will
 be false/off.
 
@@ -91,6 +91,7 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
         config.get('optimistic', DEFAULT_OPTIMISTIC))])
 
 
+#pylint: disable-msg=too-many-arguments
 class MqttSwitch(SwitchDevice):
     """ Represents a switch that can be togggled using MQTT """
     def __init__(self, hass, name, state_topic, command_topic,

